app/blueprints/fund_builder/routes.py

```diff
@@ -204,80 +208,208 @@ def clone_round(round_id, fund_id):
-        return redirect(url_for("build_fund_bp.index"))
+    if round_id:
+        round = get_round_by_id(round_id)
+        form = populate_form_with_round_data(round)
+
+    if form.validate_on_submit():
+        if round_id:
+            update_existing_round(round, form)
+            flash(f"Updated round {round.title_json['en']}")
+        else:
+            create_new_round(form)
+            flash(f"Created round {form.title_en.data}")
+        return redirect(url_for(BUILD_FUND_BP_INDEX))
 
     return render_template(
         "round.html",
         form=form,
         all_funds=all_funds_as_govuk_select_items(all_funds),
+        round_id=round_id,
+    )
+
+
+def populate_form_with_round_data(round):
+    """
+    Populate a RoundForm with data from a Round object.
+
+    :param Round round: The round object to populate the form with
+    :return: A RoundForm populated with the round data
+    """
+    round_data = {
+        "fund_id": round.fund_id,
+        "title_en": round.title_json.get("en", ""),
+        "short_name": round.short_name,
+        "opens": round.opens,
+        "deadline": round.deadline,
+        "assessment_start": round.assessment_start,
+        "reminder_date": round.reminder_date,
+        "assessment_deadline": round.assessment_deadline,
+        "prospectus_link": round.prospectus_link,
+        "privacy_notice_link": round.privacy_notice_link,
+        "contact_us_banner_json": round.contact_us_banner_json.get("en", "") if round.contact_us_banner_json else "",
+        "reference_contact_page_over_email": "true" if round.reference_contact_page_over_email else "false",
+        "contact_email": round.contact_email,
+        "contact_phone": round.contact_phone,
+        "contact_textphone": round.contact_textphone,
+        "support_times": round.support_times,
+        "support_days": round.support_days,
+        "instructions_json": round.instructions_json.get("en", "") if round.instructions_json else "",
+        "feedback_link": round.feedback_link,
+        "project_name_field_id": round.project_name_field_id,
+        "application_guidance_json": (
+            round.application_guidance_json.get("en", "") if round.application_guidance_json else ""
+        ),
+        "guidance_url": round.guidance_url,
+        "all_uploaded_documents_section_available": (
+            "true" if round.all_uploaded_documents_section_available else "false"
+        ),
+        "application_fields_download_available": "true" if round.application_fields_download_available else "false",
+        "display_logo_on_pdf_exports": "true" if round.display_logo_on_pdf_exports else "false",
+        "mark_as_complete_enabled": "true" if round.mark_as_complete_enabled else "false",
+        "is_expression_of_interest": "true" if round.is_expression_of_interest else "false",
+        "feedback_survey_config": round.feedback_survey_config,
+        "eligibility_config": round.eligibility_config.get("has_eligibility", "false"),
```

Comment:
Need to check if `round.eligibility_config` is not `None` first - am getting this error locally when editing a round that has `None` for `eligibility_config`
`line 327, in populate_form_with_round_data\n    \"eligibility_config\": round.eligibility_config.get(\"has_eligibility\", \"false\"),\nAttributeError: 'NoneType' object has no attribute 'get'", "logType": "application"}`

Reply:
Could you check whether you created a new round or tried editing an existing one? I suspect the error might be due to the fact that eligibility_config wasn’t being added when creating a new round previously: https://github.com/communitiesuk/funding-service-design-fund-application-builder/blob/main/app/blueprints/fund_builder/routes.py#L269

Now we are adding it: https://github.com/communitiesuk/funding-service-design-fund-application-builder/blob/465412a79a22eca173d9eabfab8ffa563e75f17f/app/blueprints/fund_builder/routes.py#L409